source/extensions/transport_sockets/alts/grpc_tsi.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+// Some gRPC headers contains old style cast and unused parameter which doesn't
+// compile with -Werror, ignoring those compiler warning since we don't have
+// control on those source codes. This works with GCC and Clang.
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Wold-style-cast"
```

Comment:
Is this needed for Envoy?  I'm not seeing it for other grpc includes.

Reply:
Yes, without them:

```
In file included from source/extensions/transport_sockets/alts/tsi_handshaker.cc:1:
In file included from bazel-out/k8-dbg/bin/source/extensions/transport_sockets/alts/_virtual_includes/tsi_handshaker/extensions/transport_sockets/alts/tsi_handshaker.h:10:
In file included from bazel-out/k8-dbg/bin/source/extensions/transport_sockets/alts/_virtual_includes/grpc_tsi_wrapper/extensions/transport_sockets/alts/grpc_tsi.h:12:
In file included from external/com_github_grpc_grpc/src/core/tsi/alts/handshaker/alts_tsi_handshaker.h:27:
In file included from external/com_github_grpc_grpc/src/core/tsi/alts_transport_security.h:27:
In file included from external/com_github_grpc_grpc/src/core/lib/gprpp/thd.h:32:
external/com_github_grpc_grpc/src/core/lib/gprpp/memory.h:90:56: error: unused parameter 'hint' [-Werror,-Wunused-parameter]
                   std::allocator<void>::const_pointer hint = nullptr) {
                                                       ^
external/com_github_grpc_grpc/src/core/lib/gprpp/memory.h:93:37: error: unused parameter 'n' [-Werror,-Wunused-parameter]
  void deallocate(T* p, std::size_t n) { gpr_free(p); }
                                    ^
external/com_github_grpc_grpc/src/core/lib/gprpp/memory.h:97:57: error: use of old-style cast [-Werror,-Wold-style-cast]
  void construct(pointer p, const_reference val) { new ((void*)p) T(val); }
                                                        ^      ~
external/com_github_grpc_grpc/src/core/lib/gprpp/memory.h:100:12: error: use of old-style cast [-Werror,-Wold-style-cast]
    ::new ((void*)p) U(std::forward<Args>(args)...);
           ^      ~
```